src/main/scala/cromwell/binding/command/ParameterCommandPart.scala

```diff
@@ -25,7 +26,16 @@ case class ParameterCommandPart(attributes: Map[String, String], expression: Wdl
 
   override def instantiate(declarations: Seq[Declaration], parameters: Map[String, WdlValue], functions: WdlFunctions[WdlValue] = new NoFunctions): String = {
     case class VariableNotFoundException(variable: String) extends RuntimeException(s"Could not find variable: $variable")
-    def lookup(key: String) = parameters.getOrElse(key, throw new VariableNotFoundException(key))
+
+    def resolveParameter(name: String) = Try(parameters.getOrElse(name, throw new VariableNotFoundException(name)))
+    def resolveDeclaration(lookup: ScopedLookupFunction)(name: String) = declarations.find(_.name == name) match {
+      case Some(d) => d.expression.map(_.evaluate(lookup, functions)).getOrElse(Failure(new WdlExpressionException(s"Could not evaluate declaration $d")))
+      case None => Failure(new VariableNotFoundException(name))
+    }
+    def lookup(key: String): WdlValue = {
+      val attemptedResolutions = Stream(resolveParameter _, resolveDeclaration(lookup) _) map { _(key) } find { _.isSuccess }
```

Comment:
This is probably better named `attemptedResolution` as it will be 0 or 1 resolution.